### ISSUE TITLE
SIM118 false positive in for loops

### ISSUE THREAD

**Issue by ilia-kats** [astral-sh/ruff#6450]
The SIM118 rule yields a false positive when iterating over a dict's keys. For example,
```python
test = {}
for k in test.keys():
    print(k)
```
and running `ruff check --select SIM test.py` results in
```
test.py:2:5: SIM118 [*] Use `k in test` instead of `k in test.keys()`
Found 1 error.
[*] 1 potentially fixable with the --fix option.
```
with ruff 0.0.283.

**Comment by charliermarsh** [comment]
That to me looks like a true positive, iterating over a dictionary iterates over its keys, unless I’m misinterpreting the issue?

**Comment by ilia-kats** [comment]
right, my bad, sorry. For some reason I confused `for k in test` with `for k in test.items()`. Closing

**Comment by gsakkis** [comment]
Sorry to resurrect this but the [docs](https://docs.astral.sh/ruff/rules/in-dict-keys/) mention explicitly `Checks for key-existence checks against dict.keys() calls`; there's no mentioning that this applies to loops as well. So there's at least a documentation issue. 

Although IMO this should be a separate rule, say `loop-dict-keys`. It is often more readable to iterate explicitly over the keys than the dict itself so one may want to disable this while keeping the `in-dict-keys` enabled.
